xc/device.ml

```diff
@@ -1244,8 +1244,10 @@ let clean_shutdown (task: Xenops_task.t) ~xs (x: device) =
 			try (Xenctrl.domain_getinfo xc x.frontend.domid).Xenctrl.hvm_guest
 			with _ -> false
 			in
-		try release ~xc ~xs ~hvm devs x.frontend.domid x.frontend.devid
-		with _ -> ());
+		if hvm then 
```

Comment:
While this patch may be correct, I need a bit more convincing that it is :)
It would be good to go through the places in xenops_server_xen.ml where PCI functions in device.ml are called, and:
* Confirm that pci-attach and detach are always called at the right times.
* Remove any unnecessary complexity, if needed.